runners/samza/src/main/java/org/apache/beam/runners/samza/util/PipelineJsonRenderer.java

```diff
@@ -292,4 +302,29 @@ static Map<String, Map.Entry<String, String>> buildTransformIOMap(
   private static Supplier<List<String>> getIOPValueList(Map<TupleTag<?>, PCollection<?>> map) {
     return () -> map.values().stream().map(pColl -> pColl.getName()).collect(Collectors.toList());
   }
+
+  // Reads the config to build transformIOMap, i.e. map of inputs & output PValues for each
+  // PTransform
+  public static Map<String, Map.Entry<String, String>> getTransformIOMap(Config config) {
+    checkNotNull(config, "Config cannot be null");
+    final Map<String, Map.Entry<String, String>> result = new HashMap<>();
+    final String pipelineJsonGraph = config.get(SamzaRunner.BEAM_JSON_GRAPH);
+    if (pipelineJsonGraph == null) {
+      LOG.warn(
+          "Cannot build transformIOMap since Config: {} is found null ",
+          SamzaRunner.BEAM_JSON_GRAPH);
+      return result;
+    }
+    JsonObject jsonObject = JsonParser.parseString(pipelineJsonGraph).getAsJsonObject();
+    JsonArray transformIOInfo = jsonObject.getAsJsonArray("transformIOInfo");
+    transformIOInfo.forEach(
+        transform -> {
+          final String transformName =
+              transform.getAsJsonObject().get("transformName").getAsString();
+          final String inputs = transform.getAsJsonObject().get("inputs").getAsString();
```

Comment:
Can we directly get the inputs/outputs as list here? Serde json as list, instead of manipulating the delimeter.

Reply:
done, made this return a input and output pvalues as list here